rfd/0058-package-distribution.md

```diff
@@ -0,0 +1,66 @@
+---
+authors: Fred Heinecke (fred.heinecke@goteleport.com)
+state: draft
+---
+
+# RFD 58 - Package Distribution
+
+## What
+
+Hosting and configuration of APT and YUM repositories.
+
+## Why
+
+Currently we are building .deb and .rpm file in Teleport's Drone pipeline via reprepro and createrepo, then publishing to AWS S3. Between the tools used and the pipeline's current configuration there are two main issues: a lack of channel support for major versions, and an inability to host multiple minor versions for each major version. By fixing these issues we can allow customers to upgrade the teleport package to the latest minor version release for their major version, as well as allow them to roll back to a previous minor version if need be.
+
+## Details
+
+Several potential solutions were investigated and their features compared as shown below:
+
+| Feature/Product                            | Current solution           | Current solution with fixes                                                                                  | JFrog Artifactory                                                       | PackageCloud                                   | GCP Artifact Registry                                                             |
+|--------------------------------------------|----------------------------|--------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------|------------------------------------------------|-----------------------------------------------------------------------------------|
+| Repo signing key goes to third party infra | Yes, Google (via Drone)    | Yes, Google (via Drone)                                                                                      | Yes (for SaaS)                                                          | Yes (for SaaS)                                 | Yes                                                                               |
+| Who provides signing key                   | Teleport                   | Teleport                                                                                                     | Teleport                                                                | Either                                         | Google                                                                            |
+| OOTB third party secret provider support   | AWS, Hashicorp, Kubernetes | AWS, Hashicorp, Kubernetes                                                                                   | Hashicorp                                                               | No                                             | N/A                                                                               |
```

Comment:
I'd say "out of the box" or "built in" instead of OOTB.  I had to look this up, so others will probably need to as well.

Reply:
Agreed - RFD updated.